src/callsite.jl

```diff
@@ -124,7 +124,7 @@ end
 get_mi(c::Callsite) = get_mi(c.info)
 
 # Callsite printing
-mutable struct TextWidthLimiter
+mutable struct TextWidthLimiter <: IO
```

Comment:
This is a pretty interesting change. In the long run I think `TextWidthLimiter` should be factored out into its own package (it's broadly useful). But I also recall, without remembering the details, that there were some subtleties about really supporting this as an `IO` subtype. Wish I could remember more.